src/agent/modes.js

```diff
@@ -15,9 +15,73 @@ import * as mc from '../utils/mcdata.js';
+        update: async function (agent) {
+            const bot = agent.bot;
+            const block = bot.blockAt(bot.entity.position);
+            const blockAbove = bot.blockAt(bot.entity.position.offset(0, 1, 0));
+            if (blockAbove.name === 'water' || blockAbove.name === 'flowing_water') {
+                // does not call execute so does not interrupt other actions
+                if (!bot.pathfinder.goal) {
+                    bot.setControlState('jump', true);
+                }
+            }
+            else if (this.fall_blocks.some(name => blockAbove.name.includes(name))) {
+                execute(this, agent, async () => {
+                    await skills.moveAway(bot, 2);
+                });
+            }
+            else if (block.name === 'lava' || block.name === 'flowing_lava' || block.name === 'fire' ||
+                blockAbove.name === 'lava' || blockAbove.name === 'flowing_lava' || blockAbove.name === 'fire') {
+                bot.chat('I\'m on fire!'); // TODO: gets stuck in lava
+                execute(this, agent, async () => {
+                    let nearestWater = world.getNearestBlock(bot, 'water', 20);
+                    if (nearestWater) {
+                        const pos = nearestWater.position;
+                        await skills.goToPosition(bot, pos.x, pos.y, pos.z, 0.2);
+                        bot.chat('Ahhhh that\'s better!');
+                    }
+                    else {
+                        await skills.moveAway(bot, 5);
+                    }
+                });
+            }
+            else if (Date.now() - bot.lastDamageTime < 3000 && (bot.health < 5 || bot.lastDamageTaken >= bot.health)) {
+                bot.chat('I\'m dying!');
+                execute(this, agent, async () => {
+                    await skills.moveAway(bot, 20);
+                });
+            }
+            else if (agent.isIdle()) {
+                bot.clearControlStates(); // clear jump if not in danger or doing anything else
+            }
+        }
+    },
+    {
+        name: 'cowardice',
+        description: 'Run away from enemies. Interrupts other actions.',
+        interrupts: ['all'],
+        on: true,
+        active: false,
+        update: async function (agent) {
+            const enemy = world.getNearestEntityWhere(agent.bot, entity => mc.isHostile(entity), 16);
+            if (enemy && await world.isClearPath(agent.bot, enemy)) {
+                agent.bot.chat(`Aaa! A ${enemy.name}!`);
+                execute(this, agent, async () => {
+                    await skills.avoidEnemies(agent.bot, 16);
+                });
+            }
+        }
+    },
     {
         name: 'self_defense',
-        description: 'Automatically attack nearby enemies. Interrupts other actions.',
+        description: 'Attack nearby enemies. Interrupts other actions.',
         interrupts: ['all'],
         on: true,
```

Comment:
Either self_defense or cowardice should be `on: false` by default.

Reply:
I think they should both be on, not mutually exclusive. Basically if you turn off coward he'll fallback to self defense.

Reply:
Good point I forgot that order matters in this list.